sdk/offchain/src/identifier.rs

```diff
@@ -0,0 +1,386 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::subaddress::{Subaddress, SubaddressParseError};
+use anyhow::{anyhow, Result};
+use bech32::{self, u5, FromBase32, ToBase32};
+use diem_sdk::{
+    move_types::account_address::AccountAddressParseError,
+    types::{
+        account_address::AccountAddress,
+        account_config::allowed_currency_code_string,
+        chain_id::{ChainId, NamedChain},
+    },
+};
+use std::collections::HashMap;
+use thiserror::Error;
+use url::Url;
+
+const DIEM_BECH32_VERSION: u8 = 1;
+
+/// Intent is a struct holdind data decoded from Diem Intent Identifier string
+/// https://dip.diem.com/dip-5/#format
+#[derive(Ord, PartialOrd, Eq, PartialEq, Hash, Clone)]
+pub struct Intent {
+    pub account_address: AccountAddress,
```

Comment:
Generally I would suggest avoiding making everything public and instead prefer accessors if you want to enforce some invariant.